craft_parts/sources/local_source.py

```diff
@@ -41,13 +41,16 @@ def __init__(self, *args, copy_function=file_utils.link_or_copy, **kwargs):
         self.source_abspath = os.path.abspath(self.source)
         self.copy_function = copy_function
 
-        if self._dirs.work_dir.resolve() == Path.cwd():
+        if self._dirs.work_dir.resolve() == Path(self.source_abspath):
+            # ignore parts/stage/dir if source dir matches workdir
             self._ignore_patterns.append(self._dirs.parts_dir.name)
             self._ignore_patterns.append(self._dirs.stage_dir.name)
             self._ignore_patterns.append(self._dirs.prime_dir.name)
         else:
+            # otherwise check if work_dir inside source dir
             with contextlib.suppress(ValueError):
-                rel_work_dir = self._dirs.work_dir.relative_to(Path.cwd())
+                rel_work_dir = self._dirs.work_dir.relative_to(self.source_abspath)
```

Comment:
I know I suggested source_abspath, but upon further consideration should it be resolve()d for consistency too?  I think you recently changed it to resolve?